### PR TITLE
Debug adding check1

### PR DESCRIPTION
**Pull request by MainiSuruchi**
[added 2 checks AAD connectivity and 445]

### CODE REVIEW

File: AzFilesHybrid/AzFilesHybrid.psm1 [hunk 1]
```diff
@@ -3658,6 +3658,167 @@ class CheckResult {
...
+                -ResourceGroupName $ResourceGroupName `
+                -Filter $Filter `
+                -UserName $UserName `
+                -Domain $Domain `
+                -ObjectId $ObjectId `
+                -FilePath $FilePath
+        }
+        elseif ($directoryServiceOptions -eq "AADKERB")
+        {
+            Debug-AzStorageAccountAAdKerbAuth`
```
Comment (MaximeKjaer) [inline AzFilesHybrid/AzFilesHybrid.psm1:3719]: Should we call this EntraKerbAuth now? To be in line with the official naming.

File: AzFilesHybrid/AzFilesHybrid.psm1 [hunk 2]
```diff
@@ -3658,6 +3658,167 @@ class CheckResult {
...
+                -ResourceGroupName $ResourceGroupName `
+                -Filter $Filter `
+                -UserName $UserName `
+                -Domain $Domain `
+                -ObjectId $ObjectId `
+                -FilePath $FilePath
+        }
+        elseif ($directoryServiceOptions -eq "AADDS")
+        {
+            Write-Error "This cmdlet doesnt support Microsoft Entra Domain service authentication yet, Run Debug-AzStorageAccountADDSAuth to run the AD DS authenbtication checks. Note that not all checks are expected to pass for Microsoft Entra Kerberos enabled account  "
```
Comment (MaximeKjaer) [inline AzFilesHybrid/AzFilesHybrid.psm1:3730]:
```suggestion
            Write-Error "This cmdlet does not support Microsoft Entra Domain Services authentication yet, You can run Debug-AzStorageAccountAdDsAuth to run the AD DS authentication checks instead, but note that while some checks may provide useful information, not all AD DS checks are expected to pass for a storage account with Microsoft Entra Domain Services authentication."

```

File: AzFilesHybrid/AzFilesHybrid.psm1 [hunk 3]
```diff
@@ -3658,6 +3658,167 @@ class CheckResult {
...
+                
+            } catch {
+                $checks["CheckAADConnectivity"].Result = "Failed"
+                $checks["CheckAADConnectivity"].Issue = $_
+                Write-Error "CheckAADConnectivity - FAILED"
+                Write-Error $_
+            }
+        }
+
+        Write-Error "This cmdlet doesnt support all the checks for Microsoft kerbersos authentication yet, Run Debug-AzStorageAccountADDSAuth to run the AD DS authenbtication checks. Note that not all checks are not expected to pass for Microsoft Entra Kerberos enabled account " -ErrorAction Stop
```
Comment (MaximeKjaer) [inline AzFilesHybrid/AzFilesHybrid.psm1:3816]: I suggest that we make this a Write-Host to make the error less dramatic-looking:

```suggestion
        Write-Host "This cmdlet does not support all the checks for Microsoft Entra Kerberos authentication yet, You can run Debug-AzStorageAccountAdDsAuth to run the AD DS authentication checks instead, but note that while some checks may provide useful information, not all AD DS checks are expected to pass for a storage account with Microsoft Entra Kerberos authentication."
```

File: AzFilesHybrid/AzFilesHybrid.psm1 [hunk 4]
```diff
@@ -3658,6 +3658,167 @@ class CheckResult {
...
+                Write-Error "CheckPort445Connectivity - FAILED"
+                Write-Error $_
+            }
+        }
+        if (!$filterIsPresent -or $Filter -match "CheckAADConnectivity")
+        {
+            try {
+                $checksExecuted += 1;
+                Write-Verbose "CheckAADConnectivity - START"
+                $TenantID = Get-AzContext
```
Comment (MaximeKjaer) [inline AzFilesHybrid/AzFilesHybrid.psm1:3799]: Let's maybe rename some variables here, it'll be clearer:

- `$TenantId` -> `$Context`
- `$Tenant` -> `$TenantId`

File: AzFilesHybrid/AzFilesHybrid.psm1 [hunk 5]
```diff
@@ -3658,6 +3658,167 @@ class CheckResult {
...
+                $Tenant = $TenantID.Tenant
+                $Response = Invoke-WebRequest -Method POST https://login.microsoftonline.com/$Tenant/kerberos
+                if ($Response.StatusCode == 200)
+                {
+                    $checks["CheckAADConnectivity"].Result = "Passed"
+                    Write-Verbose "CheckAADConnectivity - SUCCESS"
+                }
```
Comment (MaximeKjaer) [inline AzFilesHybrid/AzFilesHybrid.psm1:3806]: What if the response status code isn't 200? We need to have an `else` here that writes a failure to the check result. I think anything else than 200 is highly unexpected, but you never know (maybe they're on some public wifi and they're getting a HTTP 301 redirect, or something like that. Or maybe there's downtime in AAD, and they're receiving a 500 error. Or maybe their tenant doesn't exist and they're getting a 404?)

We don't need a case for each of these, we can just log the status code and that the check failed.

File: AzFilesHybrid/AzFilesHybrid.psm1 (hunk 2 above)
Comment (MainiSuruchi) [reply AzFilesHybrid/AzFilesHybrid.psm1:3730]: fixed

File: AzFilesHybrid/AzFilesHybrid.psm1 (hunk 1 above)
Comment (MainiSuruchi) [reply AzFilesHybrid/AzFilesHybrid.psm1:3719]: fixed

File: AzFilesHybrid/AzFilesHybrid.psm1 [hunk 6]
```diff
@@ -3658,6 +3658,174 @@ class CheckResult {
...
+                $Response = Invoke-WebRequest -Method POST https://login.microsoftonline.com/$TenantId/kerberos
+                if ($Response.StatusCode == 200)
+                {
+                    $checks["CheckAADConnectivity"].Result = "Passed"
+                    Write-Verbose "CheckAADConnectivity - SUCCESS"
+                }
+                else{
+                    $checks["CheckAADConnectivity"].Result = "Failed"
+                    $checks["CheckAADConnectivity"].Issue = "Expected response is 200, but we got $($Response.StatusCode)"
+                    Write-Error "Unexpected failure"
```
Comment (MaximeKjaer) [inline AzFilesHybrid/AzFilesHybrid.psm1:3811]: Just to keep the logging consistent:

```suggestion
                    Write-Error "CheckAADConnectivity - FAILED"

```

**Review by MaximeKjaer — APPROVED** [review]

**Pull request merged**